shipment.go

```diff
@@ -190,16 +192,24 @@ type createShipmentRequest struct {
 //		},
 //	)
 func (c *Client) CreateShipment(in *Shipment, accounts ...*CarrierAccount) (out *Shipment, err error) {
-	req := &createShipmentRequest{Shipment: in, CarrierAccounts: accounts}
-	err = c.post(nil, "shipments", req, &out)
+	var req createShipmentRequest
+	req.Shipment.Shipment = in
+	if len(accounts) > 0 {
+		req.Shipment.CarrierAccountIDs = []string{accounts[0].ID}
```

Comment:
This needs to extract the IDs of all accounts provided. While only one account can be included for "create and buy" to succeed the API accepts a list of IDs and limits rating to the provided accounts.  Same for `CreateShipmentWithContext`.